test/unittest/statistics/dds/StatisticsQosTests.cpp

```diff
@@ -63,20 +63,128 @@ class StatisticsFromXMLProfileTests : public ::testing::Test
     {
     public:
 
-        PublisherImpl*  get_publisher_impl()
+        efd::Publisher*  get_publisher()
         {
-            return builtin_publisher_impl_;
+            return builtin_publisher_;
         }
 
     };
 
     eprosima::fastdds::statistics::dds::DomainParticipant* statistics_participant = nullptr;
-    eprosima::fastdds::statistics::dds::PublisherImpl* statistics_publisher_impl = nullptr;
+    eprosima::fastdds::dds::Publisher* statistics_publisher = nullptr;
 
     void test_setup_XMLConfigurationForStatisticsDataWritersQoS(
             eprosima::fastdds::statistics::dds::DomainParticipant*& _statistics_participant,
-            eprosima::fastdds::statistics::dds::PublisherImpl*& _statistics_publisher_impl)
+            eprosima::fastdds::dds::Publisher*& _statistics_publisher)
     {
+        const char* xml =
+                "                                                                                                               \
+            <?xml version=\"1.0\" encoding=\"utf-8\"  ?>                                                                    \
+            <dds xmlns=\"http://www.eprosima.com/XMLSchemas/fastRTPS_Profiles\">                                            \
+            <profiles>                                                                                                      \
+                    <participant profile_name=\"statistics_participant\" is_default_profile=\"true\">                       \
+                    <rtps>                                                                                                  \
+                            <propertiesPolicy>                                                                              \
+                            <properties>                                                                                    \
+                                    <property>                                                                              \
+                                    <name>fastdds.statistics</name>                                                         \
+                                    <value>HISTORY_LATENCY_TOPIC;PUBLICATION_THROUGHPUT_TOPIC;DATA_COUNT_TOPIC;DISCOVERY_TOPIC</value>\
```

Comment:
I am missing some cases that were tested with the second XML file:

* Check that it worked without using the alias and using directly the topic name (both setting the property as the profile name being used in that specific topic).
* The `GENERIC_STATISTICS_PROFILE` is not being tested applying these changes.

Reply:
On the next commit, I have rewritten the tests in a (i think) more elegant way. Instead of creating two files and point them from the environment, I use directly the load_XML_profiles_string() so it is clearer.

Reply:
Probably when the original code was implemented, the option of loading the XML profiles directly from a string was not implemented. As you say, now that this feature is included with Fast DDS it makes more sense. Thank you!